src/libcrun/linux.c

```diff
@@ -430,16 +430,85 @@ get_bind_mount (const char *src, libcrun_error_t *err)
   return get_and_reset (&open_tree_fd);
 }
 
+static pid_t
+create_userns_for_idmapped_mount (const char *options, libcrun_error_t *err)
+{
+  cleanup_free char *dup_options = xstrdup (options);
+  char *option, *saveptr = NULL;
+  cleanup_pid pid_t pid = -1;
+  pid_t xchg_pid;
+
+  pid = syscall_clone (CLONE_NEWUSER | SIGCHLD, NULL);
+  if (UNLIKELY (pid < 0))
+    return crun_make_error (err, errno, "clone");
+
+  if (pid == 0)
+    {
+      prctl (PR_SET_PDEATHSIG, SIGKILL);
+      while (1)
+        pause ();
+      _exit (EXIT_SUCCESS);
+    }
+  for (option = strtok_r (dup_options, ";", &saveptr); option; option = strtok_r (NULL, ";", &saveptr))
+    {
+      cleanup_free char *ids = xstrdup (option + 5 /* strlen ("uids=") and strlen ("gids=") */);
+      char proc_file[64];
+      char *it;
+      int ret;
+
+      for (it = ids; *it; it++)
+        {
+          if (*it == '-')
+            *it = ' ';
```

Comment:
I am not sure if i am looking parsing here correctly or not.

For some reason in my OS `/proc/<someprocess/uid_map` columns for `User ID` and `range` is separated by `8 spaces` by default but here we are separating by single white-space.

I don't think that's going to be an issue but just adding a comment here so i can clear my doubts.

Reply:
I don't this is a issue I see write like this at other places as well.